### PR TITLE
Added working progress message

### PR DESCRIPTION
**Pull request by NickNaso**
Added a working progress message to empty documentation files.

### CODE REVIEW

File: doc/generator.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Generator
+
+## What is generator
+
+**[generator-napi-module](https://www.npmjs.com/package/generator-napi-module)** is a module to quickly generate a skeleton module using
+**N-API**, the new API for Native addons. This module automatically sets up your
+**gyp file** to use **node-addon-api**, the C++ wrappers for N-API and generates 
+a wrapper JS module. Optionally, it can even configure the generated project to 
+use **TypeScript** instead.
```
Comment (mhdawson) [inline doc/generator.md]: @sampsongao is this correct.  I was not aware it has anything special for TypeScript

File: doc/node-gyp.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+The main reason for this is that we need to link to the Node.js dependencies and 
+headers correcrtly, another reason is that we need a cross platform way to build
+C++ soucre into binary for the target platform,
+
+Until now **node-gyp** is the **de-fafto** standard build tool for writing 
+Node.js addons. It's based on Google's **gyp** build tool, which abstract away 
+many of the tedious issues related to cross platform building.
+
+**node-gyp** use a file called ```binding.gyp``` that is located on the root of
```
Comment (mhdawson) [inline doc/node-gyp.md]: use -> uses

File: doc/node-gyp.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+### **my_module.cc**
+
+```cpp
+#include <napi.h>
+
+// ...
+
+/**
+* This code is our entry-point. We receive two arguments here, the first is the
+* environment that represent an indipendent instance of the JavaScript runtime,
```
Comment (mhdawson) [inline doc/node-gyp.md]: in indipendent -> independent

File: doc/node-gyp.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+```cpp
+#include <napi.h>
+
+// ...
+
+/**
+* This code is our entry-point. We receive two arguments here, the first is the
+* environment that represent an indipendent instance of the JavaScript runtime, 
+* the sencond is exports, the same as module.exports in a .js file.
```
Comment (mhdawson) [inline doc/node-gyp.md]: sencond -> second

File: doc/node-gyp.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+* Normally you would attach properties to exports but you can use the
```
Comment (mhdawson) [inline doc/node-gyp.md]: I would change this sentence to be

You can either add properties to the exports object passed in or create your own exports object. In either case you must return the object to be used as the exports for the module when you return from the Init function.

File: doc/setup.md [hunk 6]
```diff
@@ -1,6 +1,21 @@
...
+Before starting to use **N-API** you need to assure you have the following 
+prerequisites:
+
+* **Node.JS** see: [Installing Node.js](https://nodejs.org/)
+
+* **Node.js native addon build tool** 
+
+  - **[node-gyp](node-gyp.md)**
+
+  - **[cmake-js](cmake-js.md)**
```
Comment (mhdawson) [inline doc/setup.md]: How widely used is cmake-js ? Just wondering if we should be calling it out here compared to node-gyp which is used for most modules.

File: doc/node-gyp.md (hunk 2 above)
Comment (NickNaso) [reply doc/node-gyp.md]: Ok done

File: doc/node-gyp.md (hunk 3 above)
Comment (NickNaso) [reply doc/node-gyp.md]: Ok done

File: doc/node-gyp.md (hunk 4 above)
Comment (NickNaso) [reply doc/node-gyp.md]: Ok done

File: doc/node-gyp.md (hunk 5 above)
Comment (NickNaso) [reply doc/node-gyp.md]: Do you mean this phrase?
```
...
* Normally you would attach properties to exports but you can use the 
* module argument to replace its exports property so you are exporting a single 
* thing, the equivalent of: module.exports = function () { ... }.
```

File: doc/setup.md (hunk 6 above)
Comment (NickNaso) [reply doc/setup.md]: I we consider the npm statistics yes node-gyp is the de-facto build tool for the node.js addon, but based on this issue https://github.com/nodejs/node-addon-api/issues/83 I thinked to insert some reference on documentation, maybe we can remove from **setup.md**  and maintain only on **readme.md**

File: doc/node-gyp.md (hunk 5 above)
Comment (mhdawson) [reply doc/node-gyp.md]: Yes

File: doc/setup.md (hunk 6 above)
Comment (mhdawson) [reply doc/setup.md]: ok that sounds good.

File: doc/node-gyp.md (hunk 5 above)
Comment (NickNaso) [reply doc/node-gyp.md]: Ok done

File: doc/setup.md (hunk 6 above)
Comment (NickNaso) [reply doc/setup.md]: Removed c-make-js from setup.md

File: doc/generator.md (hunk 1 above)
Comment (NickNaso) [reply doc/generator.md]: @mhdawson I just tried to use generator and at one point it ask if I want generate TypeScript wrapper. I attached the image below:
schermata 2017-11-17 alle 11 07 44

Comment (mhdawson) [reply doc/generator.md]: Ok, I was thinking of the conversion tool not the generator. I stand corrected.

**Comment by mhdawson** [comment]
This seems to have conflicts when I try to land it.  Can you squash down to 1 commit.

**Comment by NickNaso** [comment]
I'm synced with the master so now all should be fine

**Comment by mhdawson** [comment]
@NickNaso I still get

```
Falling back to patching base and 3-way merge...
Auto-merging README.md
CONFLICT (content): Merge conflict in README.md
Failed to merge in the changes.
Patch failed at 0001 Scaffolding for documentation
The copy of the patch that failed is found in:
   /home/mhdawson/newpull/land/node-addon-api/.git/rebase-apply/patch
When you have resolved this problem, run "git am --resolved".
If you prefer to skip this patch, run "git am --skip" instead.
To restore the original branch and stop patching, run "git am --abort".
```

I'm hoping squashing down to 1 will resolve.

**Comment by NickNaso** [comment]
So I need to modify README.md file and push another commit?

**Comment by mhdawson** [comment]
@NickNaso, no just squash all the commits down to a single commit. Right now it shows 44 commits.  After the squash it should show 1.  (git rebase -i HEAD~44)

**Comment by mhdawson** [comment]
The intervening commits still seem to be giving me problems.  I think if you squash down to 1 commit with a rebase it might resolve those.

**Comment by NickNaso** [comment]
@mhdawson Could you help me in doing that?

**Comment by mhdawson** [comment]
Landed via https://github.com/nodejs/node-addon-api/pull/192 instead.

**Pull request closed without merging**